lib/graphql/analysis/analyze_query.rb

```diff
@@ -43,7 +52,18 @@ def visit_analyzers(visit_type, irep_node, analyzers_and_values)
       analyzers_and_values.each do |reducer_and_value|
         reducer = reducer_and_value[0]
         memo = reducer_and_value[1]
-        next_memo = reducer.call(memo, visit_type, irep_node)
+
+        begin
+          next_memo = reducer.call(memo, visit_type, irep_node)
+        rescue GraphQL::AnalysisError => e
+          @errors << e
+        end
+
+        if next_memo.is_a?(Hash) && next_memo[:errors].present?
+          @errors << next_memo[:errors]
+          next_memo[:errors] = []
```

Comment:
I wonder if it's possible to implement this feature without touching the reducer's "private" state. I think of this flow a bit like a safe deposit box: we manage `memo` on behalf of the reducer, but it is the reducer's _own_ state, it should be reducer-defined how that object is used.